python/python/bystro/ancestry/train.py

```diff
@@ -560,6 +560,38 @@ def get_variant(x):
     return pc_loadings
 
 
+def restrict_loadings_variants_to_vcf(
+    pc_loadings: pd.DataFrame, genos: pd.DataFrame
+) -> tuple[pd.DataFrame, pd.DataFrame, int]:
+    """Restrict variant list to overlap between gnomad loadings and reference vcf."""
+    # IGSR version of 1kgp is current reference vcf
+    var_overlap = pc_loadings.index.intersection(genos.index)
+    pc_loadings_overlap = pc_loadings[pc_loadings.index.isin(var_overlap)]
+    genos_overlap = genos[genos.index.isin(var_overlap)]
+    # Ensure that genos transpose and pc_loadings have corresponding shape and vars
+    print("genos_overlap.index", genos_overlap.index)
+    print(pc_loadings_overlap.index)
+    assert genos_overlap.T.shape[1] == pc_loadings_overlap.shape[0]
+    assert set(genos_overlap.index) == set(pc_loadings_overlap.index)
+    assert (genos_overlap.columns == genos.columns).all()
+    assert "SampleID1" in genos_overlap.columns
+    # Record amount of overlap
+    num_var_overlap = len(var_overlap)
+    return pc_loadings_overlap, genos_overlap, num_var_overlap
+
+
+def apply_pca_transform(pc_loadings_overlap: pd.DataFrame, genos_overlap: pd.DataFrame) -> pd.DataFrame:
+    """Transform vcf with genotypes in dosage format with PCs loadings from gnomad PCA."""
+    # Dot product
+    transformed_data = genos_overlap.T @ pc_loadings_overlap
```

Comment:
if we're doing this correctly then transformed_data should have the right index and columns by default, is there a reason why we have to set its metadata manually below?

Reply:
this also appears to be the only time we're really using genos_overlap (as opposed to checking its properties) so would it make more sense to define genos_overlap as genos_overlap.T instead?

Reply:
There's a few places where genos_overlap is tested so it would either end up requiring a transpose step again or changing all those tests (which is easy to do but just wondering whether that's conceptually a better move than leaving the transpose step only in the place where it is needed as more of a general thought). I think setting the metadata was just a holdover from when it got converted to numpy first but no longer needed.

Reply:
There's a pretty strong convention in statistics / ML to orient matrices so that observed entities correspond to rows, and features correspond to columns.  I'd try to honor that convention.